pkg/service/node.go

```diff
@@ -220,29 +221,43 @@ func (n *NodeService) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	}
 
 	targetPath := req.GetTargetPath()
-	notMnt, err := n.fsMounter.IsLikelyNotMountPoint(targetPath)
+	notMnt, err := n.mounter.IsLikelyNotMountPoint(targetPath)
 	if err != nil && !os.IsNotExist(err) {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 	if !notMnt {
 		klog.Infof("Volume %s already mounted on %s", req.GetVolumeId(), targetPath)
 		return &csi.NodePublishVolumeResponse{}, nil
 	}
-	err = n.dirMaker.Make(targetPath, 0750)
-	// MkdirAll returns nil if path already exists
-	if err != nil {
-		return nil, err
-	}
 
 	//TODO support mount options
-	req.GetStagingTargetPath()
-	klog.Infof("Staging target path %s", req.GetStagingTargetPath())
-
-	klog.Infof("GetMount() %v", req.VolumeCapability.GetMount())
-	fsType := req.VolumeCapability.GetMount().FsType
-	klog.Infof("Mounting devicePath %s, on targetPath: %s with FS type: %s",
-		device, targetPath, fsType)
-	err = n.fsMounter.Mount(device.Path, targetPath, fsType, []string{})
+	stagingTarget := req.GetStagingTargetPath()
+	klog.Infof("Staging target path %s", stagingTarget)
+
+	fsType := ""
+	if block {
+		// Make sure target file exists for bind mount
+		f, err := os.OpenFile(targetPath, os.O_CREATE, os.FileMode(0644))
+		if err != nil {
+			if !os.IsExist(err) {
+				return nil, status.Error(codes.AlreadyExists, fmt.Sprintf("could not create bind target for block volume %s", targetPath))
+			}
+		} else {
+			_ = f.Close()
+		}
+	} else {
+		err = n.dirMaker.Make(targetPath, 0750)
```

Comment:
doesn't look like we were doing this before so why are we doing it now?

Reply:
I copied that directly from the original code, we were doing it before. I just moved it into the filesystem case because in the block device case I have to create a file for bind mounting.

Reply:
oh I see now